src/ucp/core/ucp_mm.c

```diff
@@ -444,3 +445,52 @@ ucp_mem_advise(ucp_context_h context, ucp_mem_h memh,
     UCP_THREAD_CS_EXIT(&context->mt_lock);
     return status;
 }
+
+ucs_status_t ucp_mpool_malloc(ucs_mpool_t *mp, size_t *size_p, void **chunk_p)
+{
+    ucp_worker_h worker = ucs_container_of(mp, ucp_worker_t, reg_mp);
+    ucp_mem_desc_t *chunk_hdr;
+    ucp_mem_map_params_t mem_map_params;
+    ucp_mem_h memh;
+    ucs_status_t status;
+
+    mem_map_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                                UCP_MEM_MAP_PARAM_FIELD_LENGTH |
+                                UCP_MEM_MAP_PARAM_FIELD_FLAGS;
+    mem_map_params.address    = NULL;
+    mem_map_params.length     = *size_p + sizeof(*chunk_hdr);
+    mem_map_params.flags      = UCP_MEM_MAP_ALLOCATE;
+
+    status = ucp_mem_map(worker->context, &mem_map_params, &memh);
```

Comment:
In this case I'd need to allocate memh by my own, but seems I'd need to do that, because mem_map takes ctx lock.